op-service/sources/eth_client.go

```diff
@@ -352,26 +315,31 @@ func (s *EthClient) PayloadByLabel(ctx context.Context, label eth.BlockLabel) (*
 // It verifies the receipt hash in the block header against the receipt hash of the fetched receipts
 // to ensure that the execution engine did not fail to return any receipts.
 func (s *EthClient) FetchReceipts(ctx context.Context, blockHash common.Hash) (eth.BlockInfo, types.Receipts, error) {
+	receipts, ok := s.recProvider.CachedReceipts(blockHash)
+	if ok {
+		// should always hit the cash
+		info, err := s.InfoByHash(ctx, blockHash)
+		if err != nil {
+			return nil, nil, fmt.Errorf("querying header: %w", err)
+		}
+		return info, receipts, nil
+	}
+
 	info, txs, err := s.InfoAndTxsByHash(ctx, blockHash)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("querying block: %w", err)
 	}
-	// Try to reuse the receipts fetcher because is caches the results of intermediate calls. This means
-	// that if just one of many calls fail, we only retry the failed call rather than all of the calls.
-	// The underlying fetcher uses the receipts hash to verify receipt integrity.
-	var job *receiptsFetchingJob
-	if v, ok := s.receiptsCache.Get(blockHash); ok {
-		job = v
-	} else {
-		txHashes := eth.TransactionsToHashes(txs)
-		job = NewReceiptsFetchingJob(s, s.client, s.maxBatchSize, eth.ToBlockID(info), info.ReceiptHash(), txHashes, s.rethDbPath)
-		s.receiptsCache.Add(blockHash, job)
-	}
-	receipts, err := job.Fetch(ctx)
+
+	txHashes, block := eth.TransactionsToHashes(txs), eth.ToBlockID(info)
+	receipts, err = s.recProvider.FetchReceipts(ctx, block, txHashes)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if err := validateReceipts(block, info.ReceiptHash(), txHashes, receipts); err != nil {
+		return info, nil, fmt.Errorf("invalid receipts: %w", err)
+	}
```

Comment:
We always validate receipts (this was just part of the `receiptsFetchingJob` prior to this refactor). I'm wondering if we actually want to skip this validation if `trustRPC == true`?

Reply:
Decided to skip validation if trusting the RPC. Would like to get confirmation on this being ok.